packages/bvaughn-architecture-demo/src/suspense/PauseCache.ts

```diff
@@ -36,6 +36,34 @@ export const {
   executionPoint => executionPoint
 );
 
+export { getPauseIdForExecutionPointIfCached };
+
+const pointAndTimeByPauseId = new Map<PauseId, { point: ExecutionPoint; time: number }>();
+
+export function getPointAndTimeForPauseId(pauseId: PauseId) {
+  return pointAndTimeByPauseId.get(pauseId);
+}
+
+export function getPauseIdSuspense(
+  replayClient: ReplayClientInterface,
+  point: ExecutionPoint,
+  time: number
+) {
+  const pauseId = getPauseIdForExecutionPointSuspense(replayClient, point);
+  pointAndTimeByPauseId.set(pauseId, { point, time });
+  return pauseId;
+}
+
+export async function getPauseIdAsync(
+  replayClient: ReplayClientInterface,
+  point: ExecutionPoint,
+  time: number
+) {
+  const pauseId = await getPauseIdForExecutionPointAsync(replayClient, point);
+  pointAndTimeByPauseId.set(pauseId, { point, time });
+  return pauseId;
+}
```

Comment:
Seems like maybe the changes in this file are unnecessary? I think we could just do this, couldn't we?
```js
const pointAndTimeByPauseId = new Map<
  PauseId,
  { point: ExecutionPoint; time: number }
>();

export const {
  getValueSuspense: getPauseIdForExecutionPointSuspense,
  getValueAsync: getPauseIdForExecutionPointAsync,
  getValueIfCached: getPauseIdForExecutionPointIfCached,
} = createGenericCache2<
  ReplayClientInterface,
  [executionPoint: ExecutionPoint],
  PauseId
>(
  async (client, executionPoint) => {
    const createPauseResult = await client.createPause(executionPoint);
    await client.waitForLoadedSources();
    cachePauseData(
      client,
      createPauseResult.pauseId,
      createPauseResult.data,
      createPauseResult.stack
    );

    // This is the only new line
    pointAndTimeByPauseId.set(pauseId, { point, time });

    return createPauseResult.pauseId;
  },
  (executionPoint) => executionPoint
);

export function getPointAndTimeForPauseId(pauseId: PauseId) {
  return pointAndTimeByPauseId.get(pauseId);
}
```

Reply:
> `pointAndTimeByPauseId.set(pauseId, { point, time });`

But where do we get the point/time from?

Reply:
Yeah, sorry. Realized that a minute or two after posting.